src/query/fees/fees.hooks.ts

```diff
@@ -0,0 +1,31 @@
+import { UseQueryOptions, UseQueryResult } from 'react-query';
+
+import { Estimations, TransactionFeeEstimation } from '@models/fees-types';
+import { useGetFeeEstimations } from '@query/fees/fees.query';
+import {
+  useFeeEstimationsState,
+  useFeeRateState,
+  useFeeState,
+} from '@store/transactions/fees.hooks';
+
+export function useFeeEstimationsQuery(
+  transactionPayload: string,
+  estimatedLen: number | null
+): UseQueryResult<TransactionFeeEstimation, Error> {
+  const [, setFeeEstimations] = useFeeEstimationsState();
+  const [, setFee] = useFeeState();
+  const [, setFeeRate] = useFeeRateState();
+
+  const queryOptions: UseQueryOptions<TransactionFeeEstimation> = {
+    onSuccess: (data: TransactionFeeEstimation) => {
+      if (data) {
+        setFeeEstimations(data.estimations);
+        // Set defaults to middle estimation
+        setFee(data.estimations[Estimations.Middle].fee);
+        setFeeRate(data.estimations[Estimations.Middle].fee_rate);
+      }
+    },
+  };
+  const result = useGetFeeEstimations(transactionPayload, estimatedLen, queryOptions as any);
+  return result;
```

Comment:
```suggestion
  return useGetFeeEstimations(transactionPayload, estimatedLen, queryOptions as any);
```

Reply:
O, yep, think was console logging here but will just return it now, thx.